test/os.cc

```diff
@@ -0,0 +1,87 @@
+#include "eventuals/os.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace eventuals::test {
+namespace {
+
+#if defined(__linux__) || defined(__MACH__)
+
+using eventuals::os::Thread;
+using testing::MockFunction;
+
+void foo() {
+  // Do nothing.
+}
+
+static bool functor_was_called = false;
+
+struct Moveable {
+  Moveable() = default;
+  ~Moveable() = default;
+  Moveable(const Moveable& c) = delete;
+  Moveable& operator=(const Moveable&) = delete;
+  Moveable(Moveable&&) {}
+
+  void operator()() {
+    functor_was_called = true;
+  }
+};
+
+TEST(Thread, NotJoinable) {
+  Thread t;
+  EXPECT_FALSE(t.is_joinable());
+}
+
+TEST(Thread, Joinable) {
+  Thread t{&foo, "thread_name"};
+  EXPECT_TRUE(t.is_joinable());
+  t.join();
+  EXPECT_FALSE(t.is_joinable());
+}
+
+TEST(Thread, LambdaThatCapturesEverything) {
+  MockFunction<void()> start;
+
+  EXPECT_CALL(start, Call())
+      .Times(1);
+
+  Thread t{[&]() {
+             start.Call();
+           },
+           "thread_name",
+           4'194'304};
+
+  t.join();
+}
+
+TEST(Thread, LambdaThatCapturesNothing) {
+  Thread t{[]() {}, "thread_name"};
+  t.detach();
+}
+
+TEST(Thread, FunctionPointer) {
+  Thread t1{&foo, "thread_name1"};
+  Thread t2{foo, "thread_name2"};
+
+  t1.join();
+  t2.join();
+}
+
+TEST(Thread, Moveable) {
```

Comment:
Let's simplify this test by just using a lambda and capturing a `std::unique_ptr` which can only be moved. Then we don't need the global `functor_was_called` nor the `Moveable` class.